rclpy/rclpy/node.py

```diff
@@ -205,17 +205,22 @@ def __init__(
         if parameter_overrides is not None:
             self._parameter_overrides.update({p.name: p for p in parameter_overrides})
 
-        if automatically_declare_parameters_from_overrides:
-            self._parameters.update(self._parameter_overrides)
-            self._descriptors.update({p: ParameterDescriptor() for p in self._parameters})
-
         # Clock that has support for ROS time.
         # Note: parameter overrides and parameter event publisher need to be ready at this point
         # to be able to declare 'use_sim_time' if it was not declared yet.
         self._clock = ROSClock()
         self._time_source = TimeSource(node=self)
         self._time_source.attach_clock(self._clock)
 
+        if automatically_declare_parameters_from_overrides:
```

Comment:
According to the comment above : "parameter overrides and parameter event publisher need to be ready at this point". Should we move this section of code above the ``self._clock`` line, like it originally was ?

Reply:
parameters overrides and event publisher are ready when `ROSClock()` is called, even with these lines moved.
`Node.declare_parameters()` needs `self._clock`, so I had to switch the order.

Reply:
Ah, makes sense